sklearn/base.py

```diff
@@ -437,8 +437,20 @@ def _validate_data(self, X, y=None, reset=True,
 
         return out
 
+    @property
     def _repr_html_(self):
-        """HTML representation of estimator"""
+        """HTML representation of estimator.
+
+        Used to display the HTML representation in sphinx-gallery.
+        Jupyer kernels will use `_repr_mimebundle_`.
+        """
+        if get_config()["display"] != 'diagram':
+            raise AttributeError("_repr_html_ is only defined when the "
+                                 "'display' configuration option is set to "
+                                 "'diagram'")
+        return self.__repr_html_
```

Comment:
Is the use of leading dunder intentional? I find the behavior of [private name mangling](https://docs.python.org/3.8/reference/expressions.html?highlight=mangling#atom-identifiers) potentially confusing. Maybe `self._repr_html_inner` would be more explicit?